projects/sdk-wells/src/lib/Well.ts

```diff
@@ -45,28 +56,46 @@ export class Well {
   }
 
   /**
-   * Retrieves all the well data and stores it in the Well object. Future
-   * getXXX() calls will retrive the local data instead of making an RPC call.
-   * Effectively caches data for:
-   * - getName()
-   * - getWell()
-   * - getLPToken()
-   * - getTokens()
-   * - getWellFunction()
-   * - getPumps()
-   * - getAuger()
-   * - getReserves()
-   * */
-  async loadWell() {
-    await Promise.all([this.getName(), this.getLPToken(), this.getWell(), this.getReserves()]);
+   * Loads Well data from chain
+   *
+   * If no options are specified, it will load everything. However, if
+   * an options object is passed, it will only load those the data
+   * whose options is set to true.
+   *
+   * loadWell() -- loads everything
+   * loadWell({tokens: true}) - only loads tokens
+   *
+   */
+  async loadWell(options?: LoadOptions): Promise<void> {
+    // TODO: use a multicall
+    const toLoad = [];
+
+    if (!options) {
```

Comment:
I really think this is overkill for now. IMO we should avoid stuff like this. Not blocking/required change, just a comment about our process.